### PR TITLE
Dynamic swagger.json based on handler functionality

### LINKED ISSUE #190: Swagger Generator should ignore disabled resource options in handlers
You can disable certain resource options in a handler, such as disabling create and delete here:

```javascript
var jsonApi = require("jsonapi-server");

module.exports = new jsonApi.MemoryHandler();
module.exports.create = null;
module.exports.delete = null;
```

This works as expected in the API but unfortunately the swagger generator still seems to generate documentation for these endpoints despite being disabled in an adapter. The swagger generator will need to check for the existence of a function before it generates documentation.

### PR DESCRIPTION
**Pull request by theninj4**
This resolves #190. The example server uses a handler without a `delete` function. The corresponding swagger.json should accurately reflect this.

Starting the example server and browsing to http://localhost:16006/rest/swagger.json then searching for the `/photos/{id}` path should yield just the `get` and `patch` functionality.

screen shot 2016-09-19 at 15 08 22

### CODE REVIEW

File: lib/swagger/paths.js [hunk 1]
```diff
@@ -38,85 +38,113 @@ swaggerPaths._addPathDefinition = function(paths, resourceConfig) {
...
       resourceName: relation,
       relationType: relationType,
       extraTags: resourceName,
       hasPathId: true
-    }),
-    post: swaggerPaths._getPathOperationObject({
+    });
+  }
+
+  if (resourceConfig.handlers.update) {
```
Comment (pmcnr-hx) [inline lib/swagger/paths.js]: `resourceConfig.handlers.create`

File: lib/swagger/paths.js [hunk 2]
```diff
@@ -38,85 +38,113 @@ swaggerPaths._addPathDefinition = function(paths, resourceConfig) {
...
       resourceName: relation,
       relationType: relationType,
       extraTags: resourceName,
       hasPathId: true
-    }),
-    delete: swaggerPaths._getPathOperationObject({
+    });
+  }
+
+  if (resourceConfig.handlers.update) {
```
Comment (pmcnr-hx) [inline lib/swagger/paths.js]: `resourceConfig.handlers.delete`

File: lib/swagger/paths.js (hunk 1 above)
Comment (theninj4) [inline lib/swagger/paths.js]: This is meant to be an `update` - the path handles the creation of a relationship on a resource, which is actually an `update`

File: lib/swagger/paths.js (hunk 2 above)
Comment (theninj4) [inline lib/swagger/paths.js]: As above, this is meant to be an `update`

**Comment by pmcnr-hx** [comment]
Good to go: 👍. It will be great to have the more accurate OpenAPI (aka Swagger) docs.

**Comment by theninj4** [comment]
This resolves some of #148

**Pull request merged**